src/main/java/io/github/thebusybiscuit/slimefun4/implementation/listeners/BlockPhysicsListener.java

```diff
@@ -54,17 +55,23 @@ public void onBlockFall(EntityChangeBlockEvent e) {
 
     @EventHandler
     public void onPistonExtend(BlockPistonExtendEvent e) {
-        for (Block b : e.getBlocks()) {
-            if (BlockStorage.hasBlockInfo(b) || (b.getRelative(e.getDirection()).getType() == Material.AIR && BlockStorage.hasBlockInfo(b.getRelative(e.getDirection())))) {
-                e.setCancelled(true);
-                return;
+        if(BlockStorage.check(e.getBlock()) != null) {
+            e.setCancelled(true);
+        } else {
+            for (Block b : e.getBlocks()) {
+                if (BlockStorage.hasBlockInfo(b) || (b.getRelative(e.getDirection()).getType() == Material.AIR && BlockStorage.hasBlockInfo(b.getRelative(e.getDirection())))) {
+                    e.setCancelled(true);
+                    return;
+                }
             }
         }
     }
 
     @EventHandler
     public void onPistonRetract(BlockPistonRetractEvent e) {
-        if (e.isSticky()) {
+        if(BlockStorage.check(e.getBlock()) != null) {
+            e.setCancelled(true);
+        } else if (e.isSticky()) {
```

Comment:
Put `else if` to a new line please

Reply:
Change has been performed.

Reply:
Thank you! You can mark them as resolved by pushing the "Resolve conversation" button!